### PR TITLE
Document the default poder_engsager algorithm for tmerc.

### LINKED ISSUE #2375: TM documentation is out of date
#### The documentation of the transverse Mercator projection needs to be updated
Current figure + documentation reflects the Snyder/Evenden method.  This should be supplemented (replaced?) by a figure and documentation for the Poder/Engsager method.

I will take a crack at adding the formulas.  Maybe @hobu for the figure?

### PR DESCRIPTION
**Pull request by cffk**
- [ x] Closes #2375 
 - [ x] Added clear title that can be used to generate release notes
 - [ x] Fully documented, including updating `docs/source/*.rst` for new API

Probably good to get other eyeballs on my update to the documentation to make sure that it's clear.

### CODE REVIEW

File: docs/source/operations/projections/tmerc.rst [hunk 1]
```diff
@@ -115,60 +115,211 @@ Parameters
...
+to Krüger :cite:`Krueger1912` who gave the series expansions accurate
+to :math:`n^4`, where :math:`n = (a-b)/(a+b)` is the third flattening.
+These series were extended to sixth order by Engsager and Poder in
+:cite:`Poder1998` and :cite:`Engsager2007`.  This gives full
+double-precision accuracy withing 3900 km of the central meridian
+:cite:`Karney2011tm`.
 
-:math:`\phi_0` is the latitude of origin that match the center of the map. It can be set with ``+lat_0``.
+This formulation consists of three steps: a conformal projection from
+the ellipsoid to a sphere, the spherical transversion Mercator
```
Comment (rouault) [inline docs/source/operations/projections/tmerc.rst]:
```suggestion
the ellipsoid to a sphere, the spherical transverse Mercator
```

File: docs/source/operations/projections/tmerc.rst [hunk 2]
```diff
@@ -115,60 +115,211 @@ Parameters
...
 
 Spherical form
 **************
 
-Forward projection
-==================
+Because the full (ellipsoidal) projection includes the spherical
+projection as one of the components, we present the spherial form first
+with the coordinates tagged with primes, :math:`\phi'`,
+:math:`\lambda'`, :math:`\zeta'`, :math:`x'`, :math:`y'`, so that they
```
Comment (rouault) [inline docs/source/operations/projections/tmerc.rst]: I have some trouble to see where zeta or zeta' is actually use to compute geographic or projected coordinates

File: docs/source/operations/projections/tmerc.rst [hunk 3]
```diff
@@ -115,60 +115,211 @@ Parameters
...
+can be distinguished from the corresponding ellipsoidal coordinates
+(without the primes).  The projected coordinates for the sphere are
+given by
 
 .. math::
 
-   B = \cos \phi \sin \lambda
+   \zeta' = \xi' + i\eta';\qquad x' = k_0 R \eta';\qquad y' = k_0 R \xi'
```
Comment (rouault) [inline docs/source/operations/projections/tmerc.rst]: what is i here: the imaginary unit ?

**Comment by cffk** [comment]
@rouault Thanks, I'll update to address your feedback.

File: docs/source/operations/projections/tmerc.rst [hunk 4]
```diff
@@ -115,60 +115,222 @@ Parameters
...
-  D = \frac{y}{k_0} + \phi_0
+   \zeta = \xi + i\eta;\qquad x = k_0 A \eta;\qquad y = k_0 A \xi
 
 .. math::
 
-  x' = \frac{x}{k_0}
+   A = \frac a{1+n}\biggl(1 + \frac14 n^2 + \frac1{64} n^4 +
+   \frac1{256}n^6\biggr)
+
+The series for convertion between ellipsoidal and spherical geographic
```
Comment (kbevers) [inline docs/source/operations/projections/tmerc.rst]:
```suggestion
The series for conversion between ellipsoidal and spherical geographic
```

**Review by kbevers — APPROVED** [review]

File: docs/source/operations/projections/tmerc.rst (hunk 4 above)
Comment (cffk) [reply docs/source/operations/projections/tmerc.rst]: fixed

**Comment by cffk** [comment]
@hobu I added the tmerc image you generated.  I sent this through pngtopnm | pnmtopng which converted it from RGBA to grayscale and thereby reduced its size from 342k to 169k.  Maybe this is something that should be done when you update the images as part of the CI.

**Comment by cffk** [comment]
This closes (for now!) #2375

**Comment by kbevers** [comment]
>  Maybe this is something that should be done when you update the images as part of the CI.

We actually have a script for that already: https://github.com/OSGeo/PROJ/blob/master/docs/plot/greyscale.sh. Just a matter of putting it to use I guess.

**Pull request merged**